indicators/AA21062A.yaml

```diff
@@ -0,0 +1,294 @@
+        $ua11= "Mozilla/5.0+(compatible;+YandexBot/3.0;++http://yandex.com/bots)"
+        $ua12= "Mozilla/5.0+(X11;+Linux+x86_64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/51.0.2704.103+Safari/537.36"
+        $ua13= "antSword/v2.1"
+        $ua14= "Googlebot/2.1+(+http://www.googlebot.com/bot.html)"
+        $ua15= "Mozilla/5.0+(compatible;+Baiduspider/2.0;++http://www.baidu.com/search/spider.html)"
+
+      condition:
+        any of ($ua*)
+    }
+---
+name: "RCE via OAB VD cmdlet"
+description: |
+  "OAB External Access (https://www.microsoft.com/security/blog/2021/03/02/hafnium-targeting-exchange-servers/)"
+  "A remote code execution (RCE) exploit was also observed in use against multiple organizations. This RCE appears to reside within the use of the"
+  "Set-OabVirtualDirectory ExchangePowerShell cmdlet. Evidence of this activity can be seen in Exchange's ECP Server logs."
+  "Discovered by Volexity - "https://www.volexity.com/blog/2021/03/02/active-exploitation-of-microsoft-exchange-zero-day-vulnerabilities/"
+confidence: 8
+ioc_type: "yara"
+indicator:
+  files: "C:\\Program Files\\Microsoft\\Exchange Server\\V15\\Logging\\ECP\\Server\\*.LOG"
+  rule: |
+    rule unauthenticated_execution
+    {
+      meta:
+        cve= "CVE-2021-27065"
+        description= "CVE-2021-27065 exploitation can be detected via Exchange log files"
+        reference= "https://www.microsoft.com/security/blog/2021/03/02/hafnium-targeting-exchange-servers/"
+      strings:
+
+      $req= "S:CMD=Set-OabVirtualDirectory.ExternalUrl='"
+
+      condition:
+
+      any of ($req)
+    }
+---
+name: "Possible CVE-2021-26855 Exploitation"
+description: |
+  "IoC for the exploitation of CVE-2021-26855"
+confidence: 10
+ioc_type: "events"
+indicator:
+  event_type: "Windows Powershell"
+  event.event_data: "~= *ServerInfo~*/*"
+---
+name: "CVE-2021-26858"
+description: |
+  "CVE-2021-26858 exploitation can be detected via the Exchange log file"
+  "https://www.microsoft.com/security/blog/2021/03/02/hafnium-targeting-exchange-servers/"
+confidence: 10
+ioc_type: "yara"
+indicator:
+  files: "C:\\Program Files\\Microsoft\\Exchange Server\\V15\\Logging\\OABGeneratorLog\\*.log"
+  rule: |
+    rule cve_2021_26858_detection
+    {
+      meta:
+        description= "Exploitation of CVE-2021-26858 can be detected by downloaded files located in non-standard paths"
+        source= "https://www.microsoft.com/security/blog/2021/03/02/hafnium-targeting-exchange-servers/"
+
+      strings:
+        $str1= "Download failed and temporary file"
+      condition:
+        any of ($str1)
+    }
+---
+name: "Possible CVE-2021-26857 Exploitation"
+description: |
+  "CVE-2021-26857 exploitation can be detected via the Windows Application event logs."
+  "Source - https://www.microsoft.com/security/blog/2021/03/02/hafnium-targeting-exchange-servers/"
+confidence: 8
+ioc_type: "events"
+indicator:
+  event_type: "Application"
+  event.event_data: "~= *System.InvalidCastException*"
```

Comment:
Should be `"~= .*System.InvalidCastException.*"`

Reply:
Corrections made.

thanks again for pointing these out.  It was hard to test these since event log detections aren't working yet.